### PR TITLE
:wrench: setup: for open-meme backend

### PR DESCRIPTION
**Pull request by akshay782**
Hi,
I did a setup for backend, I did look at your v1.0 and backend structure it is all in one file
so I tried to create a maintainable folder structure for the backend
**I tried to completed -** 
1. setup express server
2. create a user model (I think someone is already working on a meme model)
3. create user auth route(testing) file and their controller file
4. connect MongoDB database

.env sample
sample-env

if this is fulfilling your requirements then I will work on https://github.com/shubhamcodez/OpenMeme/issues/5

### CODE REVIEW

**Comment by akshay782** [comment]
if you see I haven't added bookmarks and liked post fields in the user model
since they can be added later

**Comment by shubhamcodez** [comment]
Code looks fine. 
Did you run it? 

If yes, please send a short video showcasing, it's functionality.
Else, I'll run it locally and merge by tomorrow.

**Comment by akshay782** [comment]
> Code looks fine. Did you run it?
> 
> If yes, please send a short video showcasing, it's functionality. Else, I'll run it locally and merge by tomorrow.

Yeah, I run it
how I can make PR before running it, 
it is not etiquette
I will try to make a video only if you find any difficulties running it

## Steps -
**First set .env variables**
1. run `npm install` in the server-side folder
2. run `npm start` in the same folder
    it will display two messages
    ```bash
       Database Connected
       Server is running on port {port that you declare in .env}...
    ```
3. then check API in postman or browser 
   URL = localhost:`port that you declare in .env`/api/get-user

File: server-side/models/user.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        unique: true,
+        index: true,
+        lowercase: true
+    },
+    email: {
+        type: String,
+        required: true,
+        unique: true,
+        trim: true,
+        unique: true,
```
Comment (devstra) [inline server-side/models/user.js:34]: This is already on line 32 👆 .

**Review by shubhamcodez — APPROVED** [review]

**Review by shubhamcodez — APPROVED** [review]

File: server-side/models/user.js (hunk 1 above)
Comment (akshay782) [reply server-side/models/user.js]: Thank you for correcting 😇

**Pull request merged**